From 22e41f75d3888962b4760328e39ae5a9b4851c88 Mon Sep 17 00:00:00 2001
From: Lin Shengwang <linshengwang1@huawei.com>
Date: Fri, 30 Sep 2022 18:08:44 +0800
Subject: [PATCH] sched/core: Change depends of SCHED_CORE
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 22e41f75d3888962b4760328e39ae5a9b4851c88
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5OOWG
CVE: NA

--------------------------------------------------------------------------

Due to the conflict between Core Scheduling and SMT expeller,
so SCHED_CORE and QOS_SCHED_SMT_EXPELLER are mutually exclusive.

Signed-off-by: Lin Shengwang <linshengwang1@huawei.com>
Reviewed-by: lihua <hucool.lihua@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/Kconfig.preempt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/Kconfig.preempt b/kernel/Kconfig.preempt
index 5876e30c5740..e62a623031ea 100644
--- a/kernel/Kconfig.preempt
+++ b/kernel/Kconfig.preempt
@@ -102,7 +102,7 @@ config PREEMPT_DYNAMIC
 
 config SCHED_CORE
 	bool "Core Scheduling for SMT"
-	depends on SCHED_SMT
+	depends on SCHED_SMT && !QOS_SCHED_SMT_EXPELLER
 	help
 	  This option permits Core Scheduling, a means of coordinated task
 	  selection across SMT siblings. When enabled -- see
-- 
2.33.0

